Client/Frontend/Browser/Playlist/VideoPlayer/VideoPlayer.swift

```diff
@@ -0,0 +1,698 @@
+            
+            if self.pictureInPictureController?.isPictureInPictureActive == false {
+                self.playerLayer.player = self.player
+            }
+        })
+        
+        notificationObservers.append(NotificationCenter.default.addObserver(forName: .AVPlayerItemDidPlayToEndTime, object: self.player.currentItem, queue: .main) { [weak self] _ in
+            guard let self = self, let currentItem = self.player.currentItem else { return }
+            
+            self.controlsView.playPauseButton.isEnabled = false
+            self.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_pause"), for: .normal)
+            self.player.pause()
+            
+            let endTime = CMTimeConvertScale(currentItem.asset.duration, timescale: self.player.currentTime().timescale, method: .roundHalfAwayFromZero)
+            
+            self.controlsView.trackBar.setTimeRange(currentTime: currentItem.currentTime(), endTime: endTime)
+            self.player.seek(to: .zero)
+            
+            self.controlsView.playPauseButton.isEnabled = true
+            self.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_play"), for: .normal)
+
+            self.showOverlays(true)
+            
+            self.next()
+        })
+        
+        let interval = CMTimeMake(value: 25, timescale: 1000)
+        self.playObserver = self.player.addPeriodicTimeObserver(forInterval: interval, queue: .main, using: { [weak self] time in
+            guard let self = self, let currentItem = self.player.currentItem else { return }
+            
+            let endTime = CMTimeConvertScale(currentItem.asset.duration, timescale: self.player.currentTime().timescale, method: .roundHalfAwayFromZero)
+            
+            if CMTimeCompare(endTime, .zero) != 0 && endTime.value > 0 {
+                self.controlsView.trackBar.setTimeRange(currentTime: self.player.currentTime(), endTime: endTime)
+            }
+        })
+    }
+    
+    private func registerPictureInPictureNotifications() {
+        if AVPictureInPictureController.isPictureInPictureSupported() {
+            pictureInPictureController = AVPictureInPictureController(playerLayer: self.playerLayer)
+            guard let pictureInPictureController = pictureInPictureController else { return }
+            
+            pictureInPictureObservers.append(pictureInPictureController.observe(\AVPictureInPictureController.isPictureInPicturePossible, options: [.initial, .new]) { [weak self] _, change in
+                self?.infoView.pictureInPictureButton.isEnabled = change.newValue ?? false
+            })
+        } else {
+            infoView.pictureInPictureButton.isEnabled = false
+        }
+    }
+    
+    private func showOverlays(_ show: Bool) {
+        self.showOverlays(show, except: [], display: [])
+    }
+    
+    private func showOverlays(_ show: Bool, except: [UIView] = [], display: [UIView] = []) {
```

Comment:
showOverlays(`false`) is hard to reason about, rename this method or params,
maybe `toggleOverlay(show: Bool)`